related_topics/alcohol_consumption/alcohol_dataset.py

```diff
@@ -0,0 +1,73 @@
+import kagglehub
+import pandas as pd
+from typing import Dict, Type, List
+import re
+import os
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AlcoholDataset:
+    _KAGGLE_DATASET_PATH: str = (
+        "thedevastator/oecd-alcohol-consumption-per-capita/versions/2"
+    )
+    _FILE_YEAR_PATTERN = re.compile(r"WHR_(\d{4}).csv")
+    _YEARS = ['2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022']
+
+    def __init__(
+        self, year_to_alchohol_time_series: Dict[str, pd.DataFrame]
+    ) -> None:
+        self._alchohol_time_series = year_to_alchohol_time_series
+        self._years = sorted(
+            list(year_to_alchohol_time_series.keys())
+        )
+
+    def get_years(self) -> List[str]:
```

Comment:
I see a lot of repeated code.
Let's create a parent class to store common code there.
Too much code will be a menace to maintain.

Reply:
Agreed!